### ISSUE TITLE
esp32s3

### ISSUE THREAD

**Issue by adriangrigore** [cpq/mdk#17]
Any plans on adding esp32s3 support?

**Comment by adriangrigore** [comment]
Solved it myself eventually! :D

**Comment by Joebeazelman** [comment]
How did you do it?

**Comment by adriangrigore** [comment]
> How did you do it?

I understand that is a nasty response, saying I did it and not release code, but it was messy, I just didn't get myself to release it. I'll take a look around my computer, see if can I find it. I think I adapted the esp32 linker script to the esp32s3 datasheet mostly.

**Comment by Joebeazelman** [comment]
Please checkout the project we're working on. It's an esp32s3 specific project. Would love your insight.

**Comment by adriangrigore** [comment]
Yeah, I used the https://github.com/apache/nuttx/tree/master/boards/xtensa/esp32s3/common/scripts linker scripts in the end. I think you can take those and fit the mdk code to them.

**Comment by Joebeazelman** [comment]
I'm not sure what we're doing wrong, but we can't get anything to work. We immediately crash right after we run it. It just spits out garbled characters on the terminal.

**Comment by sfranzyshen** [comment]
> I'm not sure what we're doing wrong, but we can't get anything to work. We immediately crash right after we run it. It just spits out garbled characters on the terminal.

can you share your work ... I also would like to see esp32s3 support added to this project ... perhaps we can all work together to figure it out ... if @adriangrigore really did succeed (...) in porting maybe they can post their work as well ...

**Comment by Joebeazelman** [comment]
> > I'm not sure what we're doing wrong, but we can't get anything to work. We immediately crash right after we run it. It just spits out garbled characters on the terminal.
> 
> can you share your work ... I also would like to see esp32s3 support added to this project ... perhaps we can all work together to figure it out ... if [@adriangrigore](https://github.com/adriangrigore) really did succeed (...) in porting maybe they can post their work as well ...

Sure! Have you had any success with it? My code is based on Ada, but the startup code is in C.

**Comment by nikolz0** [comment]
Hi, 
compiled all examples for ESP32C3 without errors,
 wrote to the chip,
 but the output to the terminal is missing in all examples. 

[examples.zip](https://github.com/user-attachments/files/20649746/examples.zip)

What's wrong?
